### PR TITLE
Fix export with new coala version

### PR DESCRIPTION
**Pull request by ptibogxiv**
@aspangaro for one customer; we need to update code to this in order to export with new coala version.

### CODE REVIEW

File: htdocs/accountancy/class/accountancyexport.class.php [hunk 1]
```diff
@@ -355,21 +355,49 @@ public function exportCogilog($objectLines)
 	 */
 	public function exportCoala($objectLines)
 	{
-		// Coala export
-		$separator = ";";
```
Comment (aspangaro) [inline htdocs/accountancy/class/accountancyexport.class.php]: Why you remove déclaration of end l'une and separator ?

Comment (ptibogxiv) [reply htdocs/accountancy/class/accountancyexport.class.php]: @aspangaro bad paste... to many things in same time :/

**Comment by aspangaro** [comment]
For the user of the old version of Koala, maybe create a new export With version in label.

In develop branch, With LDCompta export, i have specify v9 and higher for example.

You have a pdf for the new export format for documentation ?

**Review by aspangaro — APPROVED** [review]

**Comment by ptibogxiv** [comment]
i don't have this . I had some mistake with a customer ; his local dev find this fix with coala (hard to fix bug without see it in action...). He send me this; and it works with customer. I can ask them the coala docs and version.

	/**
	 * Export format : COALA
	 *
	 * @param array $objectLines data
	 *
	 * @return void
	 *
	 * KN 25062019 : Adapt Coala export format for previous Coala version requirements
	 * KN 28062019 : Disable old Coala export format
	 */
	public function exportCoala($objectLines)
	{
		/** KN 25062019 : Disable old Coala export format work-in-progress
		
		// Coala export
		$separator = ";";
		$end_line = "\n";

		foreach ( $objectLines as $line ) {
			$date = dol_print_date($line->doc_date, '%d/%m/%Y');
			print $date . $separator;
			print $line->code_journal . $separator;
			print length_accountg($line->numero_compte) . $separator;
			print $line->piece_num . $separator;
			print $line->doc_ref . $separator;
			print price($line->debit) . $separator;
			print price($line->credit) . $separator;
			print 'E' . $separator;
			print length_accountg($line->subledger_account) . $separator;
			print $end_line;
		}
		**/
		
		// Coala export
		$separator = " ";
		$end_line = "\n";
		
		// Print file export header
		print 'COMPTABILITE "cosmetique"' . $end_line;
		print 'PLAN COMPTABLE' . $end_line;
		print 'ECRITURES' . $end_line;
		print $end_line . $end_line . $end_line;
		
		// Build list of society names
		$subledger_label_list = array();
		foreach ( $objectLines as $line ) {
		    
		    if ($line->subledger_label != '') {
		        $subledger_label_list[$line->doc_ref] = $line->subledger_label;
		    }
		}

		foreach ( $objectLines as $line ) {
			
			print "\t";
			$date = dol_print_date($line->doc_date, '%d/%m/%Y');
			print $date . $separator;
			print $line->code_journal . $separator;
			
			$is_subledger_account = false;
			$subledger_account = length_accountg($line->subledger_account);
			if ($subledger_account != '') {
				print length_accountg($line->subledger_account) . $separator;
				$is_subledger_account = true;
			}
			else {
				print length_accountg($line->numero_compte) . $separator;
			}
			print '"' . $line->doc_ref . '"' . $separator;
			
			print '"' . $subledger_label_list[$line->doc_ref] . '"' . $separator;
			
			if ($line->sens=='D') {
				print 'D' . $separator;
				print str_replace(' ', '', price($line->debit,0,'',1,-1,3)) . $separator;
			}
			if ($line->sens=='C') {
				print 'C' . $separator;
				print str_replace(' ', '', price($line->credit,0,'',1,-1,3)) . $separator;
			}
			print 'E' . $separator;
			if ($is_subledger_account) {
				print 'A' . $separator;
			}
			
			print $separator;
			print $end_line;
		}

**Comment by eldy** [comment]
Yes please, ask your customer the COALA version

**Comment by klmntech** [comment]
Hi, I'm the local dev of @ptibogxiv 's customer.
The version of Coala is 19.0

**Comment by eldy** [comment]
@aspangaro do you confirm fix for coala v19 ? Do we have to include a if on version ?

**Comment by eldy** [comment]
@aspangaro Do you think fix is ok for coala v19 ? Do we have to include a if on version or adding a constant to switch from 1 behaviour to the other one ?

**Comment by eldy** [comment]
@ptibogxiv To unblock this PR, I suggest to add a new export exportCoalaV19() so we are sure to not have regression of existing exportCoala()

**Comment by eldy** [comment]
This PR is now too old, so i close it. A lot of code may have been validated since the PR and the PR may be obsolete, or too difficult to merge due to other changes. Best choice is too cancel it to avoid to keep PR with too low chances to be validated.

**Pull request closed without merging**